### ISSUE TITLE
Update to use ng-cli

### ISSUE THREAD

**Issue by samvloeberghs** [samvloeberghs/protractor-gherkin-cucumberjs-angular2#56]
To keep this project available for the community I decided to go trough with updating this project to use the CLI. This has the following consequences:

- the example application needs to get updated to work with the cli
- the following tasks need to be completed: https://github.com/samvloeberghs/protractor-gherkin-cucumberjs-angular/issues/55 https://github.com/samvloeberghs/protractor-gherkin-cucumberjs-angular/issues/48
- the configuration to have it work with **saucelabs and electron needs to keep on working**

@DennisJaamann @Fiontan

**Comment by samvloeberghs** [comment]
@Fiontan I send you an invite to be collaborator of this project. If you want to collaborate, please accept.
Let's **start from scratch** and work in a branch `update-cli`. 

I don't have much time too spare until mid June, but will help where possible.

**Comment by DennisJaamann** [comment]
@Fiontan @samvloeberghs  I believe it is the only sensible thing to do. I took the Angular class angular 2+ starter kit (with CLI) and just created a new branch to update. I have everything configured in there already, it just needs some clean up. I can share the branch later tonight if you want by pushing it to my fork remote. Let me know what you prefer

**Comment by samvloeberghs** [comment]
I prefer to start from https://github.com/angular/angular-cli#generating-and-serving-an-angular-project-via-a-development-server. Is that what you meant @DennisJaamann ?

**Comment by Fiontan** [comment]
@samvloeberghs @DennisJaamann that is exactly the route I went- it makes most sense to start from scratch with an ng new .... and then configure for use with the cucumber-specific libraries. It will change the structure somewhat to the original (Sam's) project, however, it is more aligned with the standard cli project and I think this is good for community accessibility/standardisation etc.

**Comment by DennisJaamann** [comment]
@samvloeberghs  No, I actually started from here https://github.com/AngularClass/angular-starter

And behind the scenes it does what you are mentioning. The benefit with your approach is that you cut bloat (which might not be a giant issue since you can remove the bloat very easily). 

The benefit with AngularClass is that everything is preconfigured and works out of the box. Furthermore you can easily keep this repo a community standard as well by forking / diffing with the AngularClass one.

In the end I think both options are very viable ones and it comes down to personal preference I guess. The most important thing imho is to follow a standard and stick by it, not reinvent the wheel as it will make the maintenance cost of this boilerplate project higher than it should be.

**Comment by samvloeberghs** [comment]
if we can start from scratch, using the cli approach I would prefer that.. So let's go for` ng new PROJECT-NAME` and than copy the old app into it

**Comment by Fiontan** [comment]
@samvloeberghs @DennisJaamann simply running the cli ng new will facilitate all this pre-configuration anyway...I have uploaded a fully functioning first attempt to a private bitbucket account @DennisJaamann can you provide an email please and I can give you access? It's locked with yarn too, so if you download and run the Yarn command from teh project root it all works automagically :)

**Comment by DennisJaamann** [comment]
Okay guys let's push forward then. Fiontan, I can work with you on the topic if you want to get this thing going rapidly. My bitbucket user name: DennisJaamann

**Comment by Fiontan** [comment]
@DennisJaamann you should receive a bitbucket invite anytime now. Might be worth getting it going locally and have a play around to see if you can identify any issues...enjoy and I've skyped you too :)

**Comment by DennisJaamann** [comment]
@Fiontan this looks great, should we proceed to add it to this github repo? Cheers

**Comment by Fiontan** [comment]
Hello all and apologies for the delayed response...I've been away. PLease feel free to branch it and work away.

**Comment by DennisJaamann** [comment]
I propose we make a legacy branch to keep the old code, then just copy your code to the master branch.

@samvloeberghs what do you think?

**Comment by samvloeberghs** [comment]
I prefer to keep the master as it is now, and when we're stable on the new stuff move it to there.
Just create any new branch and do the new stuff there..

**Comment by DennisJaamann** [comment]
@samvloeberghs let me rephrase: After having a stable branch and when we are able to merge, do you still want to keep the legacy code? Of course I would not want to jeopardize the pristine and working master branch right now..

**Comment by samvloeberghs** [comment]
keeping the old stuff sounds good to me :)
